cloud/shared/bin/deploy

```diff
@@ -7,7 +7,7 @@ source cloud/shared/bin/lib.sh
 cloud/shared/bin/validate_cloud_provider
 
 if [[ "${IMAGE_TAG}" == "latest" ]]; then
```

Comment:
Not blocking: Technically there's no longer any reason that this check still needs to exist since we allow deploying from latest in https://github.com/civiform/civiform-deploy/pull/14.

That said, it's still a moving tag, so we might want to continue not supporting it so that there's not drift between the label when the Git repository was checked out and when the image is being referenced.

Since our civiform-deploy script has the ability to resolve a tag to a specific SHA version of the digest, we could modify this code to ONLY accept a tag of the form `sha256:<digest>`. That would give us the benefits of knowing that our Git repository is always in sync with what's being deployed. We'd also know that the image used doesn't ever have the ability to change (due to a tag update).

The main benefit of the above is more predictability about what's actually being run
If you agree with that general sentiment, I'm happy to file a bug.

Reply:
Are you suggesting that deploy engineers use the sha hash instead of versions? Versions feel more natural.

Reply:
I'm suggesting that the "main" deployment repo only accept sha hash. Deployers would use the deploys cripts from `civiform-deploy`, which would accept versions, tags, or sha hashes.

The main thing I'm trying to get at it here is that tags / versions are immutable (the tag can move to different versions, etc), whereas a sha is immutable. We pipe this through to various downstream systems and using a floating tag for that seems like it could lead to some unintended updates. Contrast that with a sha hash, which is immutable.

Reply:
You meant that "tags / versions are mutable", right? Not immutable? 
Mentally I treat tags and versions as immutable so they correspond to hashes for me. But if that's a concern then I don't mind. It's going to be a bit less annoying to not being use latest during development, but shouldn't be that cumbersome.

How will I easily find a sha of a release? Today I go to https://hub.docker.com/r/civiform/civiform/tags Should I use its digest instead? 

Hm, also it might make debugging prod more challenging as now you can't easily see what version it is running by looking at the "image" in task definition: https://us-west-2.console.aws.amazon.com/ecs/v2/task-definitions/mikita5-td/7/json?region=us-west-2 Though there it's "latest" in real prod it will be proper version.